Convert VPopover story to MDX

sepehrrezaei opened the pull request:
## Fixes 
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Related to #584 by @dhruvkb 

## Description

<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->

This PR updates the story of VPopover to use the composition API (MDX). Here is the screenshot of the converted VPopover to MDX in the storybook:
<img width="1587" alt="Screenshot 2023-03-30 at 12 11 25" src="https://user-images.githubusercontent.com/54465154/228805461-1f31fb7d-285d-4b19-8863-e7319e3f4337.png">


## Testing Instructions

Run the Storybook from this PR to see that VContentLink is working correctly in the storybook.

## Checklist

<!-- Replace  the [ ] with [x] to check the boxes. -->

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat reviewed (changes requested):
Great start, @sepehrrezaei !
This PR needs to add `actions` to show that the Popover was opened/closed in the actions tab in the UI. You can find an example of the setup in `frontend/src/components/VHeader/meta/VContentSettingsModal.stories.mdx`. In short, you would need to add the events to the argTypes (eg. `open: { action: "open" } }`) and bind them in the template using `v-on="args"`).

We can also remove the logging functions (`onOpen`/`onClose`) and just use the actions instead.

sepehrrezaei commented:
Thanks for your explanation and example of `action`. 
I have added `open` and `close` actions to argTypes and bound them to all three stories.

dhruvkb commented on frontend/src/components/VPopover/meta/VPopover.stories.mdx:
```diff
@@ -89,7 +85,35 @@ const TwoPopoverStory = (args, { argTypes }) => ({
+
+<ArgsTable of={VPopover} />
+
+## Default VPopover
```
The 'VPopover' portion is implied in all `H2`s.

```suggestion
## Default
```

dhruvkb commented on frontend/src/components/VPopover/meta/VPopover.stories.mdx:
```diff
@@ -64,23 +60,23 @@ const ControlStory = (args, { argTypes }) => ({
-
-const TwoPopoverStory = (args, { argTypes }) => ({
-  props: Object.keys(argTypes),
+export const TwoPopoverStory = (args) => ({
```
`TwoPopoverStory` and `SinglePopoverStory` can be merged using a prop and `v-for` to generate as many popovers as needed. But that's an improvement, not a blocker.

dhruvkb commented on frontend/src/components/VPopover/meta/VPopover.stories.mdx:
```diff
@@ -89,7 +85,35 @@ const TwoPopoverStory = (args, { argTypes }) => ({
+## Default VPopover
+
+<Canvas>
+  <Story name="Default" argTypes={popoverArgTypes}>
```
If the arg types are repeated, they can be moved to the `<Meta>` tag at the top of the file.

```suggestion
  <Story name="Default" >
```

dhruvkb commented on frontend/src/components/VPopover/meta/VPopover.stories.mdx:
```diff
@@ -1,36 +1,39 @@
-    label: "text",
-    labelledBy: "text",
-  },
+<Meta title="Components/VPopover" components={VPopover} argTypes={{}} />
+
+export const popoverArgTypes = {
+  hideOnEsc: { control: { type: "boolean" } },
+  hideOnClickOutside: { control: { type: "boolean" } },
+  autoFocusOnShow: { control: { type: "boolean" } },
+  autoFocusOnHide: { control: { type: "boolean" } },
+  placement: { options: [...popoverPlacements], control: { type: "radio" } },
+  label: { control: { type: "text" } },
+  labelledBy: { control: { type: "text" } },
+  close: { action: "close" },
+  open: { action: "open" },
 }
```
Instead of repeating arg types in each story, they can be set for the entire component.

```suggestion
<Meta title="Components/VPopover" components={VPopover} argTypes={{
  hideOnEsc: { control: { type: "boolean" } },
  hideOnClickOutside: { control: { type: "boolean" } },
  autoFocusOnShow: { control: { type: "boolean" } },
  autoFocusOnHide: { control: { type: "boolean" } },
  placement: { options: [...popoverPlacements], control: { type: "radio" } },
  label: { control: { type: "text" } },
  labelledBy: { control: { type: "text" } },
  close: { action: "close" },
  open: { action: "open" },
}} />
```

dhruvkb reviewed (changes requested):
This is great, @sepehrrezaei and definitely on the right track! But I have some important changes and some optional improvements to point out.

sepehrrezaei commented:
> This is great, @sepehrrezaei and definitely on the right track! But I have some important changes and some optional improvements to point out.

Hi, Sorry for the delay! I have changed the headers and added v-for for the default story to utilize the story for many numbers that can be controlled in the controls tab as popoverItems. 
Thanks for your suggestions!

sepehrrezaei commented:
Hi, Is all the issues solved in the change request? I think I have changed all the issues that have been mentioned and committed the changes.

obulat commented on frontend/src/components/VPopover/meta/VPopover.stories.mdx:
```diff
@@ -0,0 +1,96 @@
+  template: `
+    <div>
+      <p>
+        This story is configured to log when the popover opens or closes. Inspect the console output to observe this behavior.
```
```suggestion
        This story is configured to log when the popover opens or closes. Inspect the Actions tab to observe this behavior.
```

obulat commented:
Sorry for a long delay in reviews, @sepehrrezaei. We had a lot of work moving all the code into the monorepository (here), so some reviews got postponed.

Thank you for your awesome work and finishing all the issues for the meta-issue ✨
